[Fix] Add disable for batch request limit

ZeljkoBenovic opened the pull request:
# Description

This PR adds the ability to disable `json-rpc` batch request limits, introduced in #638.

Some tools, like Blockscout are using batch requests in order to efficiently gather data from the chain.    
With current default limits, Blockscout in production, is not working as it is hitting this limits.   
Also, this protection is not needed for nodes that are not interacting with general public, for example a dedicated non-validator node that will serve only Blockscout.

To disable this limits just set the values to `0`:     
`--json-rpc-block-range-limit 0 ` and     
`--json-rpc-batch-request-limit 0`

Also, this PR fixes the issue that these values could not be fed trough config file. Right now, they can be set only with flags.     

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Set the flags `--json-rpc-block-range-limit 0 `  , `--json-rpc-batch-request-limit 0` and you should not see `Batch request length too long` error

ZeljkoBenovic committed:
added disable for bach json requests

ZeljkoBenovic committed:
added info log for debugging

ZeljkoBenovic committed:
enable setting bulk json-rpc limits from conf file

ZeljkoBenovic committed:
remove info log, rename method

0xAleksaOpacic reviewed (changes requested):
There is an issue with this PR, i will take some time today to debug it:

The scenario that does not work:
1. change hardcoded DefaultJSONRPCBlockRangeLimit to 50 (just so you don't need to wait for 1k blocks)
2. start 4 node network wait til 50+ blocks
3. trigger getLogs jsonrpc with (from: 0, to: latest), i am expecting ErrBlockRangeTooHigh but call passes

This issue was introduced with adding initJSONRPCBulkLimits.

```
if p.rawConfig.JSONRPCBatchRequestLimit != config.DefaultJSONRPCBatchRequestLimit {
		p.jsonRPCBatchLengthLimit = p.rawConfig.JSONRPCBatchRequestLimit
	}

if p.rawConfig.JSONRPCBlockRangeLimit != config.DefaultJSONRPCBlockRangeLimit {
		p.jsonRPCBlockRangeLimit = p.rawConfig.JSONRPCBlockRangeLimit
	}
```

this method is called after we setFlags. p.rawConfig.JSONRPCBatchRequestLimit will always be 0 because the value is defined in p.JSONRPCBatchRequestLimit. Because 0 is != config.DefaultJSONRPCBlockRangeLimit we are going to override p.jsonRPCBlockRangeLimit with 0. And with the functionality, you've added 0 means disabled.

Kourin1996 commented on command/server/server.go:
```diff
@@ -197,15 +197,15 @@ func setFlags(cmd *cobra.Command) {
 		&params.jsonRPCBatchLengthLimit,
 		jsonRPCBatchRequestLimitFlag,
 		defaultConfig.JSONRPCBatchRequestLimit,
```
Just curious. Is it ok that default value is not zero?

0xAleksaOpacic replied:
@zivkovicmilos I see that we approved that on this PR https://github.com/0xPolygon/polygon-edge/pull/638. But I agree, we should consider having default 0, and then if someone wants can adjust it

zivkovicmilos replied:
I don't think having `0` as the default is very smart. That PR introduced the default limit so we can have better node protection, and so people can't just make the node busy with a single request. If you want to disable this functionality as a node operator you can - I just don't think out-of-the-box configurations should enable something as dangerous as this.

For example, we have peer limits by default, but you can disable those with a special value.

0xAleksaOpacic replied:
Sounds reasonable, agree lets leave it like it is

0xAleksaOpacic replied:
@Kourin1996 please resolve so we can merge this

zivkovicmilos reviewed (approved):
Looks ok to me 💯 

Please see @Aleksao998's comment on removing that method, and we're good to go.

0xAleksaOpacic commented:
> Looks ok to me 💯
> 
> Please see @Aleksao998's comment on removing that method, and we're good to go.

Quick question, why we have to have jsonRPCBatchLengthLimit/jsonRPCBlockRangeLimit in 2 places.

```
type serverParams struct {

           rawConfig{
                JSONRPCBatchRequestLimit
                JSONRPCBlockRangeLimit
           }
           jsonRPCBatchLengthLimit
           jsonRPCBlockRangeLimit

}
```

ZeljkoBenovic committed:
fixed flags issue and added test

ZeljkoBenovic commented:
You were right @Aleksao998 ,

I've removed the method that doesn't do anything and I've removed redundant variables from the PR that introduced this functionality.

I've also added a small test, to verify that when set to `0`, there are no request limits.

0xAleksaOpacic reviewed (approved):
LGTM, thank you for fixing all comments. Double checked everything it works great now!
